Configuration/Configurator.h

```diff
@@ -37,6 +40,31 @@ bool configure (std::string fileName,
         AddressSpace::ASNodeManager *nm, ConfigXmlDecoratorFunction
         configXmlDecoratorFunction = ConfigXmlDecoratorFunction()); // 'empty' function by default.
 
+template<typename TParent, typename TChildren, typename TChildTypeId>
+void validateContentOrder(const TParent& parent, const TChildren& children, const TChildTypeId childTypeId)
```

Comment:
Deserves to remark:
1. that this check is only affecting configuration that got decorated and is expected to be transparent otherwise.
2. how does this algorithm actually work? could you add 2 or 3 phrases with an example of when it would throw?

Reply:
Every child object (referenced by its index in the child object collections attached to the parent) should have a corresponding entry in the parent's content_order collection. This is the information xsdcxx's content_order mechanism runs on.

Objects arising from decoration or from the original XML parse are treated equally: I mean the algorithm passes over both new (decoration) objects and original (parsed) objects. This is transparent for original objects because it validates as above: Original objects conform to this algorithm by definition of the content_order mechanism.

As regards when/how might it fail: The documentation contains an example of how decoration can fail. Documentation/UserDefinedRuntimeConfigurationModification.html